src/dht_proxy_server.cpp

```diff
@@ -0,0 +1,415 @@
+/*
+ *  Copyright (C) 2017 Savoir-faire Linux Inc.
+ *  Author : Sébastien Blin <sebastien.blin@savoirfairelinux.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#if OPENDHT_PROXY_SERVER
+#include "dht_proxy_server.h"
+
+#include "default_types.h"
+#include "dhtrunner.h"
+#include "msgpack.hpp"
+
+#include <chrono>
+#include <functional>
+#include <json/json.h>
+#include <limits>
+
+namespace dht {
+
+DhtProxyServer::DhtProxyServer(std::shared_ptr<DhtRunner> dht, in_port_t port)
+: dht_(dht)
+{
+    // NOTE in c++14, use make_unique
+    service_ = std::unique_ptr<restbed::Service>(new restbed::Service());
+
+    server_thread = std::thread([this, port]() {
+        // Create endpoints
+        auto resource = std::make_shared<restbed::Resource>();
```

Comment:
add OPTIONS endpoint

Reply:
can you try with your OpenDht proxy example in JS?

Reply:
Had to use this to get it work:
```c++
session->close(restbed::OK, {{"Access-Control-Allow-Methods", allowed},
                             {"Access-Control-Allow-Headers", "content-type"},
                             {"Access-Control-Max-Age", "86400"},});
```